### ISSUE TITLE
[Opinion] Excessive use of classes is little different to inline CSS.

### ISSUE THREAD

**Issue by StealthWombat** [elkarte/Elkarte#552]
This one is debatable, but I thought it was worth mentioning, since I think it's another thing inherited from SMF 2.0.x.

Take this:

```
    <div id="top_section">
        <div class="frame">
            <span id="top_section_notice" class="floatleft">';
```

Ok, it's auto for RTL if, and only if, the themer actually wants it floated in the same position as default. However, for RTL it makes no semantic sense since it will be floated right. LTR is only "normal" for some people.

The span is already defined in the CSS. Purely presentational stuff like its positioning is also better in the CSS. Yes, it will mean a bit more code in rtl.css, but it will also mean less markup, and the code in rtl.css will be easier to change.

Basically, classes and id's should only be there to tell you what something is, not how it looks in the original default layout.

ETA: This is why, when I wrote the markup and CSS orginally, I did not have a floatleft class on that span, or similar classes in several other locations. Adding them only makes things harder to customise, and complicates the markup for no real benefit.

[/pontificate] :wink:
